### PR TITLE
[#338] Added callable as Hystrix fallback

### PR DESCRIPTION
**Pull request by marcingrzejszczak**

### CODE REVIEW

File: micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientSpec.groovy [hunk 1]
```diff
@@ -281,6 +273,37 @@ class ServiceRestClientSpec extends Specification {
...
+                    .andCommandKey(HystrixCommandKey.Factory.asKey("Command"))
+            final String fallbackText = 'fallback'
+            Callable<ResponseEntity<String>> fallback = new Callable<ResponseEntity<String>>() {
+                @Override
+                ResponseEntity<String> call() throws Exception {
+                    return ResponseEntity.ok(fallbackText)
+                }
+            }
+
+
```
Comment (dst) [inline micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientSpec.groovy]: You have inconsistent formatting after spec sections.

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/request/HttpMethod.groovy [hunk 2]
```diff
@@ -61,6 +63,23 @@ interface HttpMethod<U, T> {
...
+     * Example:
+     * <code>
+     * .withCircuitBreaker(HystrixCommand.Setter.withGroupKey(HystrixCommandGroupKey.Factory.asKey("Group"))
+     *      .andCommandKey(HystrixCommandKey.Factory.asKey("Command")), {return new ResponseEntity<String>("service unavailable", HttpStatus.METHOD_FAILURE)})
+     * </code>
+     *
+     * @param setter
+     *      Fluent interface for HystrixCommand constructor arguments
+     * @param hystrixFallback
+     *      @see HystrixCommand#getFallback()
```
Comment (szpak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/request/HttpMethod.groovy]: Formatting

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/request/HttpMethod.groovy [hunk 3]
```diff
@@ -61,6 +63,23 @@ interface HttpMethod<U, T> {
...
+     */
+    @Deprecated HttpMethod<U, T> withCircuitBreaker(HystrixCommand.Setter setter, Closure hystrixFallback)
```
Comment (szpak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/request/HttpMethod.groovy]: `@Deprecated` in the previous line?

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/request/HttpMethod.groovy (hunk 2 above)
Comment (szpak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/request/HttpMethod.groovy]: Maybe a small note in javadoc why it was deprecated.

File: micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientSpec.groovy [hunk 4]
```diff
@@ -19,14 +14,11 @@ import spock.lang.AutoCleanup
...
+import java.util.concurrent.Callable
 import java.util.concurrent.Executors
 import java.util.concurrent.ScheduledExecutorService
 
-import static org.springframework.http.HttpMethod.DELETE
-import static org.springframework.http.HttpMethod.GET
-import static org.springframework.http.HttpMethod.HEAD
-import static org.springframework.http.HttpMethod.POST
-import static org.springframework.http.HttpMethod.PUT
+import static org.springframework.http.HttpMethod.*
```
Comment (szpak) [inline micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientSpec.groovy]: Here comes the starts again :).

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/request/HttpMethod.groovy [hunk 5]
```diff
@@ -74,5 +93,5 @@ interface HttpMethod<U, T> {
      * @param hystrixFallback
      *      @see HystrixCommand#getFallback()
      */
-    HttpMethod<U, T> withCircuitBreaker(HystrixCommand.Setter setter, Closure hystrixFallback)
+    HttpMethod<U, T> withCircuitBreaker(HystrixCommand.Setter setter, Callable hystrixFallback)
```
Comment (szpak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/request/HttpMethod.groovy]: `Callable<T>`?

**Comment by szpak** [comment]
Besides comments, +1

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/request/HttpMethod.groovy (hunk 5 above)
Comment (marcingrzejszczak) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/request/HttpMethod.groovy]: It depends :)

**Pull request merged**